src/utils/auth_utils.js

```diff
@@ -42,11 +42,117 @@ const NPOBackend = axios.create({
   withCredentials: true,
 });
 
+const refreshUrl = `https://securetoken.googleapis.com/v1/token?key=${process.env.REACT_APP_FIREBASE_APIKEY}`;
+
+// Sets a cookie in the browser
+const setCookie = (key, value, config) => {
+  let cookie = `${key}=${value}; max-age=${config.maxAge}; path=${config.path}`;
+
+  if (config.domain) {
+    cookie += `; domain=${config.domain}`;
+  }
+  if (cookie.secure) {
+    cookie += '; secure';
+  }
+  document.cookie = cookie;
```

Comment:
what was the reasoning for doing this rather than using react cookies? just FYI, we fixed the bug with cookies not working in safari. TLDR, adding you can only add a secure cookie if you are using https. but in localhost, we are using http on safari. we updated dev for `cookie config` right [here](https://github.com/ctc-uci/authentication-frontend/blob/dev/src/utils/cookie_utils.js)

Reply:
Ah in that case, please just update this method for checking if the url is http or https similar to the link in my first comment